modules/opengl/src/shader/shader.cpp

```diff
@@ -38,6 +38,9 @@ namespace inviwo {
 
 Shader::Shader(const std::vector<std::pair<ShaderType, std::string>> &items, Build buildShader)
     : id_{glCreateProgram()}, warningLevel_{UniformWarning::Ignore} {
+    if (std::find_if(std::begin(items), std::end(items), [](const auto& item) { return item.first == ShaderType::Vertex;}) == std::end(items)) {
+        throw Exception("Vertex shader required, provide for example img_identity.vert");
```

Comment:
Good catch. Wrapped the id in a scoped variable

Reply:
This won't work as intended! In principle one can define multiple vertex shaders. But they must contain exactly one main function when combined... I'd assume there will be a GLSL link error due to missing main.

Reply:
I think that would be a different error. Here we just check that one did not miss providing a vertex shader. I guess we could add a search for a main function but it would make it a bit more complicated and a bit out of scope for the PR?

Reply:
yes, I think so, too.